### PR TITLE
Add a changelog

### PR DESCRIPTION
**Pull request by nvnieuwk**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+- Added a new [documentation site](https://nextflow-io.github.io/nf-validation/). (@ewels and @mashehu)
+- Removed the `file-path-exists`, `directory-path-exists` and `path-exists` and added a [`exists`](https://nextflow-io.github.io/nf-validation/nextflow_schema/nextflow_schema_specification/#exists) parameter to the schema. (@mirpedrol)
+- New [`errorMessage`](https://nextflow-io.github.io/nf-validation/nextflow_schema/nextflow_schema_specification/#errormessage) parameter for the schema which can be used to create custom error messages. (@mirpedrol)
+- Samplesheet validation now happens in `validateParameters()` using the schema specified by the `schema` parameter in the parameters schema. (@mirpedrol)
+
+### Improvements
+
+- The `meta` maps are now immutable by default, see [`ImmutableMap`](https://nextflow-io.github.io/nf-validation/samplesheets/immutable_map/) for more info (@nvnieuwk)
+- `validateAndConvertSamplesheet()` has been renamed to `fromSamplesheet()`
```
Comment (mirpedrol) [inline CHANGELOG.md:21]:
```suggestion
- `validateAndConvertSamplesheet()` has been renamed to `fromSamplesheet()`
- Refactor `--schema_ignore_params` to `--validationSchemaIgnoreParams`
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# nextflow-io/nf-validation: Changelog
+
```
Comment (mirpedrol) [inline CHANGELOG.md:2]:
```suggestion

## Version 0.3.0 (dev)

### Bug fixes

- Only validate a path if it is not null ([#50](https://github.com/nextflow-io/nf-validation/pull/50/files))
- Only validate a file with a schema if the file path is provided ([#51](https://github.com/nextflow-io/nf-validation/pull/51/files))

```

**Review by mirpedrol — APPROVED** [review]
LGTM, thanks! :)

I have suggested the changed for the current unreleased version

**Review by mashehu — APPROVED** [review]
No need to do it retro-actively, but I prefer if PRs are mentioned along with the new features (like it is in 0.3.0dev). It makes it easier to quickly find out more about the rationale and implementation of the feature.

**Comment by nvnieuwk** [comment]
> No need to do it retro-actively, but I prefer if PRs are mentioned along with the new features (like it is in 0.3.0dev). It makes it easier to quickly find out more about the rationale and implementation of the feature.

Thank you! I'll keep that in mind when adding new features!

**Comment by ewels** [comment]
@nvnieuwk / @mirpedrol / @mashehu - if someone could please copy these over to the GitHub releases, that would be great 👍🏻 

Thanks!

**Comment by nvnieuwk** [comment]
> @nvnieuwk / @mirpedrol / @mashehu - if someone could please copy these over to the GitHub releases, that would be great 👍🏻
> 
> Thanks!

On it!

**Comment by ewels** [comment]
Thanks!

**Pull request merged**